frontend/src/component/feature/FeatureView/FeatureView.tsx

```diff
@@ -231,144 +278,228 @@ export const FeatureView = () => {
 
     return (
         <div ref={ref}>
-            <StyledHeader>
-                <StyledInnerContainer>
-                    <StyledFlagInfoContainer>
-                        <FavoriteIconButton
-                            onClick={onFavorite}
-                            isFavorite={feature?.favorite}
-                        />
-                        <div>
-                            <StyledFlagInfoContainer>
-                                <StyledFeatureViewHeader data-loading>
-                                    {feature.name}{' '}
-                                </StyledFeatureViewHeader>
-                                <Tooltip
-                                    title={
-                                        isFeatureNameCopied
-                                            ? 'Copied!'
-                                            : 'Copy name'
-                                    }
-                                    arrow
-                                >
-                                    <IconButton
-                                        onClick={handleCopyToClipboard}
-                                        style={{ marginLeft: 8 }}
+            {flagOverviewRedesign ? (
```

Comment:
Is the header coupled with a lot of other things on the page? Maybe it makes sense to refactor it into a separate file?

Reply:
It's definitely couple to a few fair few things (which is why I did it inline initially instead of as a standalone function). Happy to take a look at it in a follow-up, though.